generic.go

```diff
@@ -41,6 +58,44 @@ func Compare[T any](a, b T, less LessFn[T]) int {
 	return 0
 }
 
+// NextPowerOf2 is a fast calculation implementation of 2^x.
+// see: https://stackoverflow.com/questions/466204/rounding-up-to-next-power-of-2
+// go:inline
```

Comment:
Ideally we wouldn't have these `go:inline` comments. Are they really necessary for good performance?

Reply:
fixed